### PR TITLE
[azopenai] Readme and examples

### PR DESCRIPTION
**Pull request by richardpark-msft**
Creating examples and a readme for azopenai.

### CODE REVIEW

File: sdk/cognitiveservices/azopenai/README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Azure OpenAI client module for Go
 
-Azure OpenAI is a managed service that allows developers to deploy, tune, and generate content from OpenAI models on Azure resources.
+NOTE: this client can be used with Azure OpenAI and Open AI.
```
Comment (chlowell) [inline sdk/cognitiveservices/azopenai/README.md:3]: Should there be a space?
```suggestion
NOTE: this client can be used with Azure OpenAI and OpenAI.
```

File: sdk/cognitiveservices/azopenai/client_shared_test.go [hunk 2]
```diff
@@ -74,10 +72,10 @@ func init() {
 		chatCompletionsModelDeployment = "gpt-4"
 		openAIChatCompletionsModel = "gpt-4"
 	} else {
-		if err := godotenv.Load(); err != nil {
-			fmt.Printf("Failed to load .env file: %s\n", err)
-			os.Exit(1)
-		}
+		// if err := godotenv.Load(); err != nil {
```
Comment (chlowell) [inline sdk/cognitiveservices/azopenai/client_shared_test.go:75]: Do you want to keep this code?

File: sdk/cognitiveservices/azopenai/example_client_createimage_test.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
```
Comment (chlowell) [inline sdk/cognitiveservices/azopenai/example_client_createimage_test.go:2]:
```suggestion
//go:build go1.18
// +build go1.18

// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.
```

File: sdk/cognitiveservices/azopenai/example_client_createimage_test.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+	if azureOpenAIKey == "" || azureOpenAIEndpoint == "" {
+		fmt.Fprintf(os.Stderr, "Skipping example, environment variables missing\n")
+		return
+	}
+
+	keyCredential, err := azopenai.NewKeyCredential(azureOpenAIKey)
+
+	if err != nil {
+		panic(err)
```
Comment (chlowell) [inline sdk/cognitiveservices/azopenai/example_client_createimage_test.go:30]:
```suggestion
		// TODO: handle error
```

File: sdk/cognitiveservices/azopenai/example_client_createimage_test.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+		resp, err := http.Head(*generatedImage.URL)
+
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Fprintf(os.Stderr, "Image generated, HEAD request on URL returned %d\n", resp.StatusCode)
+	}
+
+	// Output:
```
Comment (chlowell) [inline sdk/cognitiveservices/azopenai/example_client_createimage_test.go:62]: Does this really add value as an executable test? I'm thinking the ceremony of making it in principle executable just adds complexity to the docs, and we want a recorded test to actually exercise the API.

File: sdk/cognitiveservices/azopenai/client_shared_test.go (hunk 2 above)
Comment (richardpark-msft) [reply sdk/cognitiveservices/azopenai/client_shared_test.go]: Just did a little update - I needed it and incorrectly thought I didn't :)

File: sdk/cognitiveservices/azopenai/example_client_createimage_test.go (hunk 5 above)
Comment (richardpark-msft) [reply sdk/cognitiveservices/azopenai/example_client_createimage_test.go:65]: I think it's worthwhile - I use it and it makes sure that I didn't do something funky like leave out a required parameter, etc..

File: sdk/cognitiveservices/azopenai/example_client_createimage_test.go (hunk 4 above)
Comment (richardpark-msft) [reply sdk/cognitiveservices/azopenai/example_client_createimage_test.go]: Done.

File: sdk/cognitiveservices/azopenai/example_client_test.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+package azopenai_test
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/cognitiveservices/azopenai"
+)
+
+func ExampleNewClientForOpenAI() {
+	// NOTE: this constructor creates a client that connects to the public OpenAI endpoint.
```
Comment (jhendrixMSFT) [inline sdk/cognitiveservices/azopenai/example_client_test.go:12]: I think this comment should be moved to above the call to `azopenai.NewClientForOpenAI`

**Review by jhendrixMSFT — APPROVED** [review]

File: sdk/cognitiveservices/azopenai/example_client_test.go (hunk 6 above)
Comment (richardpark-msft) [reply sdk/cognitiveservices/azopenai/example_client_test.go]: Oops, yeah, that was weird.

**Pull request merged**